simulariumio/file_converter.py

```diff
@@ -19,24 +21,207 @@
+            buffer_data = FileConverter.update_trajectory_info_version(buffer_data)
         self._data = TrajectoryData.from_buffer_data(buffer_data)
 
+    @staticmethod
+    def _read_binary_file(input_file: InputFileData) -> Dict[str, Any]:
+        """
+        Read a .simularium binary file and return multiple views of the data
+        """
+        result = {}
+        with open(input_file.file_path, "rb") as open_binary_file:
+            result["byte_view"] = open_binary_file.read()
+            result["n_values"] = int(
+                len(result["byte_view"]) / BINARY_SETTINGS.BYTES_PER_VALUE
+            )
+            result["int_view"] = struct.unpack(
+                f'<{result["n_values"]}i', result["byte_view"]
+            )
+            result["float_view"] = struct.unpack(
+                f'<{result["n_values"]}f', result["byte_view"]
+            )
+        return result
+
+    @staticmethod
+    def _binary_header(binary_data: Dict[str, Any]) -> Dict[str, Any]:
+        """
+        Parse header of data from a .simularium binary file
+        """
+        pos = len(BINARY_SETTINGS.HEADER)
+        binary_id = binary_data["byte_view"][:pos].decode("utf-8")
+        if binary_id != BINARY_SETTINGS.HEADER:
+            raise DataError("Binary file is not in .simularium format")
+        pos = int(pos / BINARY_SETTINGS.BYTES_PER_VALUE)
+        header_length = binary_data["int_view"][pos]
+        header_n_values = int(
+            (header_length - len(BINARY_SETTINGS.HEADER))
+            / BINARY_SETTINGS.BYTES_PER_VALUE
+        )
+        header_values = list(binary_data["int_view"][pos : pos + header_n_values])
+        binary_version = header_values[1]
+        if binary_version != BINARY_SETTINGS.VERSION:
+            raise DataError(
+                "Binary file parsing only implemented for "
+                f"version > 1 up to current version {BINARY_SETTINGS.VERSION}, "
+                f"found {binary_version}"
+            )
+        return {
+            "n_blocks": header_values[2],
+            "block_offsets": header_values[3::3],
+            "block_types": header_values[4::3],
+            "block_lengths": header_values[5::3],
+        }
+
+    @staticmethod
+    def _binary_block_header(
+        block_index: int,
+        binary_header: Dict[str, Any],
+        binary_data: Dict[str, Any],
+    ) -> Tuple[int, int]:
+        """
+        Parse block header of data from a .simularium binary file
+        """
+        block_offset = int(
+            binary_header["block_offsets"][block_index]
+            / BINARY_SETTINGS.BYTES_PER_VALUE
+        )
+        block_type = binary_data["int_view"][block_offset]
+        if block_type != binary_header["block_types"][block_index]:
+            raise DataError(
+                "Block type is not consistent for block "
+                f"#{block_index} : {block_type} != "
+                + binary_header["block_types"][block_index]
```

Comment:
in these error messages it won't be clear which number came from the header and which came from the block